Buffs the Pirate Ship to be bomb resistant

keronshb opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Title, also makes a few computers/machines indestructible and buffs turret health.  It also allows turrets to be repairable now.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Bomber man bad.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added repairable turrets
add: Adds Tiny Fans to the pirate ship
tweak: changes some reinforced windows to plastinanium pirate windows
tweak: made the pirate shuttle + turrets bomb resistant
tweak: made most pirate machines + consoles indestructible
tweak: increased pirate turret health
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

keronshb committed:
Buffs the Pirate Ship to be bomb proof

Title, also makes a few computers/machines indestructible and buffs turret health.  It also allows turrets to be repairable now.

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -348,6 +348,22 @@
 		qdel(cover)	//deletes the cover - no need on keeping it there!
 
+//turret healing
+/obj/machinery/porta_turret/attackby(obj/item/I, mob/living/user, params)
```
Use the `welder_act` proc instead.

Ghommie commented on code/game/objects/structures/window.dm:
```diff
@@ -595,6 +595,30 @@
 	anchored = FALSE
 
+//pirate ship windows
+/obj/structure/window/plastitaniumpirate
```
Make this a subtype of `/obj/structure/window/plastitanium` and remove any duplicate variable statement on this path.

Ghommie commented on code/game/turfs/simulated/wall/mineral_walls.dm:
```diff
@@ -303,3 +303,64 @@
+
+/////////////////////Pirate Plastitanium walls/////////////////////
+
+/turf/closed/wall/mineral/plastitaniumpirate
```
Make this a subtype of `/turf/closed/wall/r_wall/syndicate`, just for the description.

Ghommie commented on code/game/turfs/simulated/wall/mineral_walls.dm:
```diff
@@ -303,3 +303,64 @@
+	fixed_underlay = list("space"=1)
+	explosion_block = 50
+
+/turf/closed/wall/mineral/plastitaniumpirate/explosive/ex_act(severity)
```
Oh, remove this, please.

Ghommie reviewed (changes requested):
Also you removed two couple directional windows from where the incomplete artillery mounts are, potentially exposing the area to depressurized space. Please fix that.

keronshb committed:
Fixes some files and pathings for this

Adds proper subtypes, put the health down a lil bit for the turret, fixed the welder type, added some tiny fans to the ship and also added some plastitanium windows where r-windows were

Ghommie commented on code/game/objects/effects/spawners/structure.dm:
```diff
@@ -163,6 +163,11 @@ again.
+//plastitanium pirate window
+
+/obj/effect/spawner/structure/window/plastitanium/pirate
+	explosion_block = 50
```
Remove this line, it's a spawner and its explosion_block value is totally meaningless.

Ghommie commented on code/game/objects/structures/window.dm:
```diff
@@ -595,6 +595,14 @@
+//pirate ship windows
+/obj/structure/window/plastitanium/pirate
+	desc = "Yarr this window be explosion proof!"
+	explosion_block = 50
```
Bring this down to 20, they shouldn't be more bomb proof than the syndicate's

Ghommie commented on code/game/turfs/simulated/wall/mineral_walls.dm:
```diff
@@ -303,3 +303,19 @@
+
+/////////////////////Pirate Plastitanium walls/////////////////////
+
+/turf/closed/wall/mineral/plastitanium/pirate
```
I told you to make this a subtype of `/turf/closed/wall/r_wall/syndicate`.
Also remove the `explosion_block` value of 50, the parent type's 20 will suffice.

Ghommie commented on code/game/turfs/simulated/wall/mineral_walls.dm:
```diff
@@ -303,3 +303,19 @@
+/turf/closed/wall/mineral/plastitanium/nosmooth/pirate/
+	explosion_block = 50
+
+/turf/closed/wall/mineral/plastitanium/overspace/pirate/
```
Also you'll fuck up the smoothing if you don't move the `/pirate` path before `/overspace`, `/nosmooth` and `/nodiagonal`. 
Refer to https://github.com/Citadel-Station-13/Citadel-Station-13/blob/2acb52fb56a7ba8e3706f5e08bd515c3620f3588/code/game/turfs/simulated/wall/mineral_walls.dm#L261 for types' values.

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -670,7 +686,11 @@
+	explosion_block = 50
+	max_integrity = 260
+	integrity_failure = 20
+
```
This will affect other energy turrets outside the pirate ship ones, please make a new subtype for them space seafolks.

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -670,7 +686,11 @@
 	lethal_projectile_sound = 'sound/weapons/laser.ogg'
-	desc = "An energy blaster auto-turret."
+	desc = "An energy blaster auto-turret. Use a welder to fix."
+	explosion_block = 50
```
Change this with armor values:
```
	armor = list("melee" = 50, "bullet" = 30, "laser" = 30, "energy" = 30, "bomb" = 50, "bio" = 0, "rad" = 0, "fire" = 90, "acid" = 90)
```
That's the base turret armor plus 20 extra points of bomb armor.

keronshb committed:
Should fix the rest of the requested changes

Title

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -27,7 +27,7 @@
 	max_integrity = 160		//the turret's health
 	integrity_failure = 80
-	armor = list("melee" = 50, "bullet" = 30, "laser" = 30, "energy" = 30, "bomb" = 30, "bio" = 0, "rad" = 0, "fire" = 90, "acid" = 90)
+	armor = list("melee" = 50, "bullet" = 30, "laser" = 30, "energy" = 30, "bomb" = 50, "bio" = 0, "rad" = 0, "fire" = 90, "acid" = 90)
```
move the armor changes to the `/obj/machinery/porta_turret/syndicate/energy/pirate` path.

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -670,7 +686,8 @@
 	lethal_projectile = /obj/item/projectile/beam/laser
 	lethal_projectile_sound = 'sound/weapons/laser.ogg'
-	desc = "An energy blaster auto-turret."
+	desc = "An energy blaster auto-turret. Use a welder to fix."
```
Undo the desc change and instead add this line above the welder_act()
```
/obj/machinery/porta_turret/examine(mob/user)
	. = ..()
	if(obj_integrity < max_integrity)
		. += "<span class='notice'>Use a welder to fix it.</span>"
```

Ghommie commented on code/game/objects/effects/spawners/structure.dm:
```diff
@@ -163,6 +163,9 @@ again.
 
+//plastitanium pirate window
+
+/obj/effect/spawner/structure/window/plastitanium/pirate
```
Add back this line or it'll spawn normal plastitanium windows.
```
	spawn_list = list(/obj/structure/grille, /obj/structure/window/plastitanium/pirate)
```

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+	explosion_block = 30
+	canSmoothWith = list(/turf/closed/wall/r_wall/syndicate/pirate, /obj/machinery/door/airlock/shuttle, /obj/machinery/door/airlock, /obj/structure/window/plastitanium/pirate, /obj/structure/shuttle/engine, /obj/structure/falsewall/plastitanium)
+
+/turf/closed/wall/r_wall/syndicate/pirate/nodiagonal/
```
Remove the last slash from the line.

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+	icon_state = "map-shuttle_nd"
+	explosion_block = 30
+
+/turf/closed/wall/r_wall/syndicate/pirate/nosmooth/
```
Same.

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+	icon_state = "wall"
+	explosion_block = 30
+
+/turf/closed/wall/r_wall/syndicate/pirate/overspace/
```
Idem.

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+/turf/closed/wall/r_wall/syndicate/pirate/nodiagonal/
+	smooth = SMOOTH_MORE
+	icon_state = "map-shuttle_nd"
+	explosion_block = 30
```
Remove, it's already the same as the parent type's

Ghommie replied:
Remove, `/turf/closed/wall/r_wall/syndicate/pirate`'s `explosion_block` value is already 30.

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+/turf/closed/wall/r_wall/syndicate/pirate/nosmooth/
+	icon = 'icons/turf/shuttle.dmi'
+	icon_state = "wall"
+	explosion_block = 30
```
Same

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+/turf/closed/wall/r_wall/syndicate/pirate/overspace/
+	icon_state = "map-overspace"
+	fixed_underlay = list("space"=1)
+	explosion_block = 30
```
Idem

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -348,6 +348,22 @@
+			return TRUE
+	else
+		to_chat(user, "<span class='notice'>The turret doesn't need repairing.</span>")
+	return
```
Remove this line.

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -348,6 +348,22 @@
 		qdel(cover)	//deletes the cover - no need on keeping it there!
 
+//turret healing
+/obj/machinery/porta_turret/welder_act(mob/living/user, obj/item/I)
```
add a `. = TRUE` line below.

Ghommie commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -348,6 +348,22 @@
+			obj_integrity = max_integrity
+			user.visible_message("[user.name] has repaired [src].", \
+								"<span class='notice'>You finish repairing the turret.</span>")
+			return TRUE
```
Remove this line.

keronshb committed:
Fixes round 3? 4?

Should fix the last round of requests?

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+/turf/closed/wall/r_wall/syndicate/pirate/nosmooth
+	icon = 'icons/turf/shuttle.dmi'
+	icon_state = "wall"
+	explosion_block = 30
```
Same.

Ghommie commented on code/game/turfs/simulated/wall/reinf_walls.dm:
```diff
@@ -258,3 +258,25 @@
+/turf/closed/wall/r_wall/syndicate/pirate/overspace
+	icon_state = "map-overspace"
+	fixed_underlay = list("space"=1)
+	explosion_block = 30
```
Same.

keronshb committed:
30 removal

Here it is

Ghommie committed:
Update reinf_walls.dm
